src/compiler/parser.ts

```diff
@@ -83,7 +83,8 @@ module ts {
                     visitNodes(cbNodes, node.modifiers) ||
                     visitNodes(cbNodes, (<SignatureDeclaration>node).typeParameters) ||
                     visitNodes(cbNodes, (<SignatureDeclaration>node).parameters) ||
-                    visitNode(cbNode, (<SignatureDeclaration>node).type);
+                    visitNode(cbNode, (<SignatureDeclaration>node).type) ||
+                    visitNode(cbNode, (<SignatureDeclaration>node).typePredicate);

```

Comment:
I would not do this.  "typePredicate" is just another way of specifying the type of the thing.  I would instead add ```interface TypePredicateNode : TypeNode { name: Identifier, type: TypeNode }```